### ISSUE TITLE
Search String has odd behavior

### ISSUE THREAD

**Issue by FatWaterAudio** [dotnet/AspNetCore.Docs#18470]
Search String does not seem to work ... in Pages/Movies/Index.cshtml.cs:
breakpoint after 
movies = movies.Where(s => s.Title.Contains(SearchString));
shows the variable movies does in fact contain the results to match input search string 

however breakpoint after
Movie = await movies.ToListAsync();
shows Movie does not contain the results to match input search string.  It contains all movie titles  

is there a buffer somewhere that needs flushing?

also I have noticed that the webpage fails on the first try after any code edit .. usually successful on second or third try .. is this related? 
   

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cc54dbf8-e6fe-a212-b2cd-c015a6ba5439
* Version Independent ID: b96b0b44-3e3a-a6c2-6b2b-c7bab4b78f0d
* Content: [Add search to ASP.NET Core Razor Pages](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/search?view=aspnetcore-2.1)
* Content Source: [aspnetcore/tutorials/razor-pages/search.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/tutorials/razor-pages/search.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
If you're using SQLite, it's case sensitive

The tutorial works. I'm not sure what step you missed. You might try asking on StackOverflow,.

**Comment by Rick-Anderson** [comment]
BTW, we've had well over a million folks tell us this series works.

**Comment by FatWaterAudio** [comment]
Hmmm .. well I am not making this stuff up .. my experience with this so far is that the environment/instructions/toolchain  one or more of them seem buggy or inconsistent.   Typically need to try at least twice to get the debugger to build/run the app.  The particular testing tasks given in the tutorial seem to fail about 10% of the time... expected results do not happen/compiler seems to ignore the new edit.    The code work is not that complicated to follow.. so not suspecting the code as much as the environment.   On other compilers or environments I have worked with,  getting the toolchain tuned for the task was where most of the startup time went.  I am new to Razor/EF style web ...sooo ... kind of in startup mode with this ... not completely convinced all of my environment variables are correct.       

 

I am running win7 with SP1. Plenty hard drive / plenty of cpu speed / plenty of mem. 

Tried changing App URL in project properties to see if the web page loads more consistently  ... not sure yet if it worked.  

The search string issue seems to have solved itself ... on the following edit to the OnGetAsync method to add Genre,  the problem simply went away .  tried commenting out new code to replicate the earlier issue ... but instead it now seems to work as intended.  Its weird buggy behavior ... I am trying to choose a webhosting suite for a commercial endeavor.   Hoping this environment can work for me.  

 

 

 

Thanks

Jim Collins

Fatwater Audio

425-212-8756 (text is best)
